backend/src/main/java/instantPolls/controller/WSController.java

```diff
@@ -20,22 +28,56 @@
 	
 	@MessageMapping("/poll/{roomId}/enter")
 	@SendTo("/room/{roomId}/users")
-    public SimpleMessage enter(@DestinationVariable String roomId) {
-        Room r = roomStorage.findRoomById(roomId);
-        r.incrementUsers();
-        SimpleMessage m = new SimpleMessage(Integer.toString(r.getNumberOfUsers()));
-        return m;
+    public SimpleMessage enter(@DestinationVariable String roomId, @Payload String userId) {
+		Room room = roomStorage.findRoomById(roomId);
+		room.getUsers().add(userId);
+		SimpleMessage message = new SimpleMessage(Integer.toString(room.getNumberOfUsers()));
+        return message;
+    }
+	
+	@MessageMapping("/poll/{userId}/allQuestions")
+	@SendTo("/user/{userId}/allQuestions")
+    public ArrayList<HashMap<String,Object>> sendQuestions(@DestinationVariable String userId,@Payload String roomId) {
+		Room room = roomStorage.findRoomById(roomId);
+		return room.getListOfQuestionsWithVotes(userId);
     }
 	
 	@MessageMapping("/poll/{roomId}/exit")
 	@SendTo("/room/{roomId}/users")
-	public SimpleMessage exit(@DestinationVariable String roomId) {
+	public SimpleMessage exit(@DestinationVariable String roomId, @Payload String userId) {
 		Room r = roomStorage.findRoomById(roomId);
 		SimpleMessage m = null;
 		if(r != null) {
-			r.decrementUsers();
-	        m = new SimpleMessage(Integer.toString(r.getNumberOfUsers()));
+			r.getUsers().remove(userId);
+			m = new SimpleMessage(Integer.toString(r.getNumberOfUsers()));
 		}
         return m;
 	}
+	
+	@MessageMapping("/poll/{roomId}/addQuestion/{questionType}")
+	@SendTo("/question/{roomId}")
+	public QuestionMessage addQuestion(@DestinationVariable String roomId, @DestinationVariable String questionType, @Payload QuestionMessage message) {
+		Question question = null;
+		QuestionMessage messageToSend = new QuestionMessage();
+		if(questionType.equals("yesNo")) {
+			question = new YesNoQuestion(message.getQuestion());
+			messageToSend.setQuestion(question.getQuestion());
+			messageToSend.setType("yesNo");
+			messageToSend.setAnswers(question.getOptions());
+			messageToSend.setNumberOfVotes(question.getNumberOfVotes());
+		} 
```

Comment:
this yesNo, in string form, is in a code at least three times. Problem of connecting question identifier from message with java code should be approached differently.